Clang7 + CUDA(clang): '__GLIBCXX__' is not defined, evaluates to 0

Error:
```
/include/alpaka/exec/ExecGpuCudaRt.hpp:148:7: error: '__GLIBCXX__' is not defined, evaluates to 0 [-Werror,-Wundef]
#if (!__GLIBCXX__) && (!BOOST_COMP_MSVC)
```
Setup:
```bash
spack load --dependencies cmake
spack load --dependencies boost@1.65.1
spack load --dependencies llvm@7.0.0
module list
cmake -DALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE=OFF -DALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE=OFF -DALPAKA_ACC_CPU_B_TBB_T_SEQ_ENABLE=OFF -DALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE=OFF -DALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE=OF -DALPAKA_ACC_CPU_BT_OMP4_ENABLE=OFF -DALPAKA_ACC_GPU_CUDA_ENABLE=ON -DALPAKA_CUDA_COMPILER=clang ..
make helloWorld
```
Edit: Used CUDA9.2

--- a/include/alpaka/exec/ExecGpuCudaRt.hpp
+++ b/include/alpaka/exec/ExecGpuCudaRt.hpp
@@ -145,7 +145,7 @@ namespace alpaka
         public:
 // gcc-4.9 libstdc++ does not support std::is_trivially_copyable.
 // MSVC std::is_trivially_copyable seems to be buggy (last tested at 15.7).
-#if (!__GLIBCXX__) && (!BOOST_COMP_MSVC)
+#if (!BOOST_COMP_MSVC) && !(defined(__GLIBCXX__) && (__GLIBCXX__))
             static_assert(
                 meta::Conjunction<
                     std::is_trivially_copyable<

